Docs fixing and cleanup (#1217)
netket/netket: docs/docs/hilbert.md

## The `DiscreteHilbert` interface

{class}`DiscreteHilbert` is also an abstract class from which any hilbert space with countable (or discrete) local degrees of freedom must inherit.
Examples of such spaces are spins or bosons on a lattice.

You can always probe their {attr}`~DiscreteHilbert.shape`, which returns a tuple 

> with the size of the hilbert space on every site/degree of freedom.

Reviewer: ```suggestion
with the size of the Hilbert space on every site/degree of freedom.
```

It should be now evident why NetKet distinguishes locally discrete/countable spaces from arbitrary (e.g: continuous) spaces: if we can index the local basis, we can perform many optimisations and write efficient kernels to compute matrix elements of operators, but also Monte-Carlo samplers will propose transitions in a very different way than in continuous spaces.

You can also obtain the total size of the hilbert space by invoking {attr}`~DiscreteHilbert.n_states`, which in general is equivalent to calling `np.prod(hi.shape)`.

```python
>>> hi.n_states
96
```

> Do bear in mind that this attribute only works if the hilbert space is indexable ({attr}`~DiscreteHilbert.is_indexable`), which is True when it has a size smaller than $ 2^{64} $. 

Reviewer: ```suggestion
Bear in mind that this attribute only works if the Hilbert space is indexable ({attr}`~DiscreteHilbert.is_indexable`), which is true if it has a dimension smaller than $2^{64}$. 
```
`Hilbert.size` is the number of sites in NetKet, so `dimension` is probably the better term here, right?
